doc/nrf/releases_and_maturity/migration/nRF54H20_migration_2.7/migration_guide_2.4.99-cs3_to_2.7_application.rst

```diff
@@ -0,0 +1,569 @@
+.. _migration_cs3_to_to_2_7_app:
+
+Migrate your application for the nRF54H20 DK to |NCS| v2.7.0 (for v2.4.99-cs3 users)
+####################################################################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+This document describes the changes you should be aware of when migrating your application from |NCS| v2.4.99-cs3 to |NCS| v2.7.0.
+
+
+Overview
+********
+
+..
+   ##TODO link to sysbuild and hwmodelv2 migration guides
+
+|NCS| v2.7.0 introduced a series of changes that might affect your existing applications.
+The following is a summary of the most important ones:
+
+Updated |NCS| toolchain
+  The |NCS| toolchain has been updated.
+  See :ref:`transition_cs3_to_2_7_env` for more info on how to upgrade your current |NCS| installation based on version 2.4.99-cs3.
+
+Sysbuild
+  A new build system was recently introduced in |NCS|
+  For more information see ``linktosysbuild_migration``
+
+Hardware Model v2
+  A new hardware model was recently introduced in |NCS|
+  For more information see ``linktohwmodelv2_migration``
+
+DTS changes
+  The layout of DTS files and the names of DTS nodes related to the updated board names have been also updated, also affecting overlay files from applications and samples.
+  If your application required a specific custom board, you must update the custom board files to match the changes done to the nRF54H20 SoC DTS files.
+
+SDFW and SCFW
+  The Secure Domain Firmware (SDFW) and System Controller Firmware (SCFW) are no longer built from the source during the application build process, but they are installed as binaries from the provided firmware bundle and programmed when flashing the application to the device.
+  For additional details, see :ref:`transition_cs3_to_2_7_env_bringup`.
+
+nRF Util has now replaced nRF Command Line Tools.
+  nRF Util is now the main command line backend utility.
+
+Updated boards
+  SOC1-based boards have been removed and FP1-based board have been added.
+  The board names for the Application, Radio, and PPR core have been updated.
+
+Lifecycle State changes
+  To correctly operate the nRF54H20 DK, it must be set in LCS ``RoT``.
+
+  Also, it is no longer possible to perform an unauthenticated backward LCS transitions.
+  This means that once set to ``RoT``, it is no longer possible to revert to LCS state ``EMPTY``.
+
+.. note::
+   Not all the features supported in |NCS| v2.4.99-cs3 have been ported to v2.7.0.
+   Additional features will be provided in future releases.
+
+Required changes
+****************
+
+The following changes are mandatory to make your application work in the same way as in previous releases.
+
+This section describes the changes related to samples and applications.
+
+Samples and applications
+========================
```

Comment:
CoreMark
--------

Several changes have been made to migrate the :ref:`coremark_sample` sample from the nRF54 customer sampling release v2.4.99-cs3 to |NCS| v2.7.0:

* The sample has been moved from the :file:`<ncs_path>/zephyr/samples/benchmarks/coremark` folder in the Zephyr repository to the :file:`<ncs_path>/nrf/samples/benchmarks/coremark` folder in the main |NCS| repository.
* The build system has been aligned to the hardware model v2.
  The ``SB_CONFIG_IMAGE_2_BOARD`` and ``SB_CONFIG_IMAGE_3_BOARD`` Kconfig options have been replaced with the ``SB_CONFIG_APP_CPUNET_RUN`` and ``SB_CONFIG_APP_CPUPPR_RUN`` Kconfig options.
  The new Kconfig options more clearly indicate on which cores the sample will be run.
  See the :file:`sysbuild.cmake` file to find out how these options are used to determine the board target.
* Because the |NCS| v2.7.0 does not support ARM Coresight System Trace Macrocell (STM) logging for the nRF54 device, STM logging has been removed from the sample.
  The sample now uses usual UART logging that allows for sending logs from only one core for each UART instance.
  The nRF54 device has only two UART instances, so the sample now can be run on two cores at most.
  The sample is always run on application core and depending on configuration can be run on either the radio core or the PPR core.
  See the ``SB_CONFIG_APP_CPUNET_RUN`` and ``SB_CONFIG_APP_CPUPPR_RUN`` Kconfig options for more details.
* The DTS overlays have been aligned to the |NCS| v2.7.0.
* The :file:`system_nrf.h` library has been included explicitly in the :file:`main.c` file to print the CPU frequency.
* The ``SB_CONFIG_PARTITION_MANAGER`` Kconfig option has been disabled in the :file:`sysbuild.conf` file to avoid conflicts with the Partition Manager.
* The :kconfig:option:`CONFIG_APP_MODE_FLASH_AND_RUN` Kconfig option has been made promptless and enabled for the PPR core.
  Currently, the PPR core does not have access to buttons and thus, the :kconfig:option:`CONFIG_APP_MODE_FLASH_AND_RUN` Kconfig option must be enabled for this core to run the benchmark.
* To make the sample run on the PPR core, pass the following build-time arguments to the build system:
  ``-DSB_CONFIG_APP_CPUNET_RUN=n -DSB_CONFIG_APP_CPUPPR_RUN=y -Dcoremark_SNIPPET=nordic-ppr-xip``.
  The ``coremark_SNIPPET`` argument is set to make the application core start the PPR core.
  Alternatively, you can build the sample from the :file:`sample.yaml` file using the following command:
  .. code-block:: console
    west build -p -b nrf54h20dk/nrf54h20/cpuapp -T sample.benchmark.coremark_ppr_xip .

Reply:
Please hold on for now - I'm waiting for @divipillai rereview. I'll let you know once it's finished.

Reply:
Review is finished. It's ready to be added to this PR.

Reply:
Please hold on for now - @kapi-no plans to make some changes. I'll let you know once it's finished.